cli/lsp/documents.rs

```diff
@@ -494,7 +488,19 @@ impl Document {
   }
 
   pub fn maybe_navigation_tree(&self) -> Option<Arc<tsc::NavigationTree>> {
-    self.0.maybe_navigation_tree.clone()
+    self.0.maybe_navigation_tree.lock().clone()
+  }
+
+  pub fn try_set_navigation_tree(
```

Comment:
`update_navigation_tree_for_version`? Also it seems racy - the version check might succeed, but then if the mutex lock is currently held, another call to this function could be made for newer version. Then it's a coin flip which call gets hold of the mutex first?

Reply:
Nope, it should be ok. I'll update the method name.